src/GraphQL.ApiTests/GraphQL.approved.txt

```diff
@@ -2473,6 +2448,23 @@ namespace GraphQL.Types.Relay.DataObjects
 }
 namespace GraphQL.Utilities
 {
+    public sealed class AppliedDirectivesValidationVisitor : GraphQL.Utilities.ISchemaNodeVisitor
```

Comment:
Well, I wrote both `AppliedDirectivesValidationVisitor` and `SchemaValidationVisitor` as classes for internal use. This is the internal schema validation logic that no one will call anywhere else. I keep forgetting about custom schems implementation. To be honest, I don't like that because of the existence of the `ISchema` interface, you are saying that these classes should be `public`. More precisely, the logic of your reasoning is clear to me, but I do not like the fact that this **obliges** us to make `public`, in fact, internal classes. I believe that this is initially the wrong path and the longer we move along it, the worse.